packages/headless/src/controllers/commerce/search/pagination/headless-search-pagination.ts

```diff
@@ -0,0 +1,31 @@
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {executeSearch} from '../../../../features/commerce/search/search-actions';
+import {commerceSearchReducer as commerceSearch} from '../../../../features/commerce/search/search-slice';
+import {loadReducerError} from '../../../../utils/errors';
+import {
+  buildCorePagination,
+  Pagination,
+} from '../../pagination/core/headless-core-commerce-pagination';
+
+/**
+ * Creates a `Pagination` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @returns A `Pagination` controller instance.
+ * */
+export function buildSearchPagination(engine: CommerceEngine): Pagination {
+  if (!loadSearchPagination(engine)) {
+    throw loadReducerError;
+  }
+
+  return buildCorePagination(engine, {
+    fetchResultsActionCreator: executeSearch,
+  });
+}
+
+function loadSearchPagination(
+  engine: CommerceEngine
+): engine is CommerceEngine {
```

Comment:
This is a bit sneaky when you're not fluent in typescript. Not returning anything will give a falsy value 👀

Reply:
Why return anything at all. This function will always return `true`

Reply:
This is a common pattern throughout the repository. I'd raise this in the #searchheadlessviplounge slack channel.